library/core/src/main/java/com/google/android/exoplayer2/extractor/mp4/AtomParsers.java

```diff
@@ -882,6 +886,12 @@ private static void parseVideoSampleEntry(ParsableByteArray parent, int atomType
         HevcConfig hevcConfig = HevcConfig.parse(parent);
         initializationData = hevcConfig.initializationData;
         out.nalUnitLengthFieldLength = hevcConfig.nalUnitLengthFieldLength;
+      } else if (childAtomType == Atom.TYPE_dvcC || childAtomType == Atom.TYPE_dvvC) {
```

Comment:
Quick question: is it intentional that this branch doesn't check that mimeType is null? I am wondering if we expect the containing loop to parse any other boxes like avcC or hvcC, and whether dvcC/dvvC is guaranteed to come after the others.

Reply:
If this is intentional, could you also explain how fallback works? Is it just for profiles 0 and 2, and do we need to do anything to the access units to make them compatible with non-DV decoders?

Reply:
In all DV streams, avcC and hcCC boxes are guaranteed to come before dvcC/dvvC boxes. non-DV decoders will just ignore the NALUs thats not known to them. and it will only take the base layer and not Dolby Vision metadata.

Reply:
Is it the case that none of the supported profiles (4, 5, 8, 9) support backwards compatibility? Just checking that we don't need to implement falling back to an AVC or HEVC decoder in the renderer.

Reply:
Maybe I didn't the question properly. Profile 4,8,9 all are backward compatible profiles. Only Profile 5 is non-backward compatible. If a DV decoder is present, then all these profiles should be decoded through DV(video/dolby-vision or its variant) decoder. If no DV decoder present profile 4 and 8 can be send through "video/hevc" decoder and profile 9 can be send through "video/avc" decoder

Reply:
Is it the case that after this change, devices that could previously play media in DV profiles 4, 8 and 9 but lack a DV decoder will no longer be able to play the streams? If so, I think we should restrict to profile 5 for now, until we have a working fallback path. If you agree I can take care of this when I merge the change internally (no need to modify anything here).

Reply:
I agree that we should keep only profile 5 in this change. Feel free to make the change. But can you also let me know the fallback that you have in mind? I was thinking of calling the MediaCodecUtil.getDecoderInfos(MimeType, false); to see if we have dv decoders on the platform before assigning the mime-type for proile 4,8,9. Will that work?

Reply:
Not sure yet but I think we should probably put this fallback in the renderer or in MediaCodecUtil (to avoid having the extractor's behavior depend on the device decoding capabilities). The codec selector could return a non-DV decoder for those profiles if it detects that no DV decoder is present.